tests/Integration/PrestaShopBundle/Routing/Linter/NamingConventionLinterTest.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace Tests\Integration\PrestaShopBundle\Routing\Linter;
+
+use PrestaShopBundle\Routing\Linter\Exception\NamingConventionException;
+use PrestaShopBundle\Routing\Linter\NamingConventionLinter;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+use Symfony\Component\Routing\Route;
+use Tests\Resources\Controller\TestController;
+
+class NamingConventionLinterTest extends KernelTestCase
+{
+    /**
+     * @var NamingConventionLinter
+     */
+    private $namingConventionLinter;
+
+    protected function setUp()
+    {
+        self::bootKernel();
+
+        $this->namingConventionLinter = self::$kernel->getContainer()
```

Comment:
Is it really needed ? The parsing logic we want to test here seems unit testable 🤔 ?

Reply:
Hmmm, well I want to use real dependencies (in this case `ControllerNameParser`) as controllers are defined in different formats, so I want to make sure I did use that dependency correctly inside.

Reply:
I can understand that but then you can create a ControllerNameParser instance 😉 and mock the [required kernel for constructor](https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/FrameworkBundle/Controller/ControllerNameParser.php#L30).

You'll only need to mock `$bundle = $this->kernel->getBundle()` to provide the right bundle for it. Then it's unit-with-1-dependency.

Reply:
Sounds like a plan, will do. :+1: